internal/testutil/assert/assert.go

```diff
@@ -0,0 +1,147 @@
+package assert
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+type miniTB interface {
+	Helper()
+	Fatalf(string, ...interface{})
+}
+
+// formatSuffix builds an optional suffix from msgAndArgs
+func formatSuffix(msgAndArgs ...interface{}) string {
+	switch len(msgAndArgs) {
+	case 0:
+		return ""
+	case 1:
+		return " - " + fmt.Sprintf("%+v", msgAndArgs[0])
+	default:
+		return " - " + fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+}
+
+// Comparable types (numbers, strings, pointers to the same object, etc.).
+func Equal(tb miniTB, got, want interface{}, msgAndArgs ...interface{}) {
+	tb.Helper()
+	if got != want {
+		suffix := formatSuffix(msgAndArgs...)
+		tb.Fatalf("got %v; want %v%s", got, want, suffix)
+	}
+}
+
+// interface{}thing else (slices, maps, structs with slices...).
+func DeepEqual(tb miniTB, got, want interface{}, msgAndArgs ...interface{}) {
```

Comment:
Usually test asserter are like this


```suggestion
func Equal(tb miniTB, want, got interface{}, msgAndArgs ...interface{}) {
	tb.Helper()
	if got != want {
		suffix := formatSuffix(msgAndArgs...)
		tb.Fatalf("got %v; want %v%s", got, want, suffix)
	}
}

// interface{}thing else (slices, maps, structs with slices...).
func DeepEqual(tb miniTB, want, got interface{}, msgAndArgs ...interface{}) {
```